pageserver/src/control_plane_client.rs

```diff
@@ -12,25 +14,31 @@ use utils::{
 
 use crate::config::PageServerConf;
 
-// Backoffs when control plane requests do not succeed: compromise between reducing load
-// on control plane, and retrying frequently when we are blocked on a control plane
-// response to make progress.
-const BACKOFF_INCREMENT: f64 = 0.1;
-const BACKOFF_MAX: f64 = 10.0;
-
 /// The Pageserver's client for using the control plane API: this is a small subset
 /// of the overall control plane API, for dealing with generations (see docs/rfcs/025-generation-numbers.md)
-pub(crate) struct ControlPlaneClient {
+pub struct ControlPlaneClient {
     http_client: reqwest::Client,
     base_url: Url,
     node_id: NodeId,
     cancel: CancellationToken,
 }
 
+#[async_trait::async_trait]
+pub trait ControlPlaneGenerationsApi {
+    async fn re_attach(&self) -> anyhow::Result<HashMap<TenantId, Generation>>;
+    async fn validate(
+        &self,
+        tenants: Vec<(TenantId, Generation)>,
+    ) -> anyhow::Result<HashMap<TenantId, bool>>;
+}
+
+unsafe impl Send for ControlPlaneClient {}
+unsafe impl Sync for ControlPlaneClient {}
```

Comment:
Why?

Reply:
There was some Arc'ish stuff going on with the interface for mocking deletion queue.  I never liked it.  Refactored it to generics in 20a3a9be7 and this goes away too.